### ISSUE TITLE
Auto-LoD issues

### ISSUE THREAD

**Issue by Alayan-stk-2** [supertuxkart/stk-code#3793]
As the PR has already been merged by @samuncle I'll list here some outstanding issues in the merged code : 
- Some code and comment is confusing. See this : 
```
    // The track has high complexity so we decrease the draw distance by 10%
    if (complexity > 3000 )
    {
        dist += (dist/10);
    }
```

The comment says "decrease the draw distance" while the code applies a +10% increase to the variable "dist". This is very unclear. It appears that what this code do is not changing the draw distance, but changing the distance at which the object is considered to be, with higher distance thus triggering a lower level of detail earlier.
- I suspect the LoD computation is inconsistent depending on scale. See this : "    `m_area *= scale;`". Area increase quadratically with scale, so area 1 scale 2 is the same as area 4 scale 1, but with this code they will get a different m_area. This should be scale*scale for consistency, but this would also require retesting the values to make sure they behave well.
- The non-smooth transition for max_draw is still present ; with a sudden jump from 250 to 265. I suggested 235 + (max_draw * 0.06) to avoid this ; 250 + (max_draw * 0.06) doesn't solve this quirk I pointed out.
- autoComputeLevel has the same incorrect confusing use of "draw distance" in the comments and code (lower values mean less culling but the draw distance comments imply the opposit)
- The geometry level used to determine auto-LoD agressivity is not linked to the graphics level preset
- While the popping generated by the lower level of auto-LoD is noticeable, in my tests with my (weak) discrete graphics card the performance benefit is unnoticeable in all tracks but Black Forest where there is a tiny benefit. I can tell a popping difference but no performance difference between the 3 geometry level with my intel iGPU (it has a cap at 60fps so at graphics 1 or 2 720p I get locked 60fps even in black forest, and increasing the graphics settings to get a lower FPS doesn't produce a clear performance difference)

**Comment by hiker** [comment]
I've already asked samuncle to move it back to a branch, it's too soon to be included in the upcoming release

**Comment by samuncle** [comment]
So, let me explain.

1) I disabled the autolodcompute for the RC. I'm sorry for any inconvenience I might have caused
2) I will fix the smooth transition for  235 + (max_draw * 0.06)  however this isn't really noticable since this only concern objects in the distance. It doesn't really matter, it's more to provide a more or less gradual LoD level. It's no better than arbitrarily putting distance in a model in blender. The cool part it's fully automated
3) draw_distance represent the maximum distance an object can be drawn (based on it's screen size). The level are then computed based on how many level you have in your LoD
4) for the link between geometry detail and preset it's no my fault it was an existing option. The option was not much used until now and it kinds of tie nicely together.
5) 
> "but changing the distance at which the object is considered to be, with higher distance thus triggering a lower level of detail earlier."
True, I just wanted an easy way to trigger this without necesseraly having to do complicated math.
6) Regarding performances, I think It needs a bit of tuning but most importantly it will shine when we will improve hidden object management (currently 10 000 of hidden objects make stk slow) which should not happen. With Auria's experimental branch + my autolod I got 89 FPS in total in black forest which is quite impressive considering the amount of objects it has.

My assumption is by making LoD creation easy + improving the performance of our scene manager we can have a huge boost in performances.
